server/src/test/java/org/opensearch/search/querygroup/tracking/QueryGroupResourceUsageTrackerServiceTests.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.querygroup.tracking;
+
+import org.opensearch.action.search.SearchShardTask;
+import org.opensearch.action.search.SearchTask;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.tasks.resourcetracker.TaskResourceUsage;
+import org.opensearch.search.querygroup.QueryGroupLevelResourceUsageView;
+import org.opensearch.search.querygroup.QueryGroupTask;
+import org.opensearch.search.querygroup.tracker.QueryGroupResourceUsageTrackerService;
+import org.opensearch.search.resourcetypes.ResourceType;
+import org.opensearch.tasks.CancellableTask;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskManager;
+import org.opensearch.tasks.TaskResourceTrackingService;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.threadpool.TestThreadPool;
+import org.opensearch.threadpool.ThreadPool;
+import org.junit.After;
+import org.junit.Before;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class QueryGroupResourceUsageTrackerServiceTests extends OpenSearchTestCase {
+    TestThreadPool threadPool;
+    TaskManager taskManager;
+    TaskResourceTrackingService mockTaskResourceTrackingService;
+    QueryGroupResourceUsageTrackerService queryGroupResourceUsageTrackerService;
+
+    @Before
+    public void setup() {
+        taskManager = new TaskManager(Settings.EMPTY, threadPool, Collections.emptySet());
```

Comment:
Shouldn't this threadPool be initialized before the `TaskManager` ?

Reply:
good catch !